stl/inc/xutility

```diff
@@ -4262,19 +4262,41 @@ public:
         _STL_VERIFY(_Right._Mystate != _State::_Nonthing, "common_iterator can only be assigned if it holds a value");
 #endif // _ITERATOR_DEBUG_LEVEL != 0
         if (_Mystate == _State::_Holds_iter && _Right._Mystate == _State::_Holds_iter) {
+            if constexpr (!is_nothrow_copy_assignable_v<const _Iter&>) {
+                _Mystate = _State::_Nothing;
+            }
             _Myiter = _Right._Myiter;
+            if constexpr (!is_nothrow_copy_assignable_v<const _Iter&>) {
+                _Mystate = _State::_Holds_iter;
+            }
         } else if (_Mystate == _State::_Holds_sentinel && _Right._Mystate == _State::_Holds_sentinel) {
+            if constexpr (!is_nothrow_copy_assignable_v<const Se&>) {
+                _Mystate = _State::_Nothing;
+            }
             _Mysentinel = _Right._Mysentinel;
+            if constexpr (!is_nothrow_copy_assignable_v<const Se&>) {
+                _Mystate = _State::_Holds_sentinel;
+            }
         } else if (_Right._Mystate == _State::_Holds_iter) {
             if (_Mystate == _State::_Holds_sentinel) {
+                if constexpr (!is_nothrow_destructible_v<_Se>) {
+                    _Mystate = _State::_Nothing;
+                }
                 _Mysentinel.~_Se();
+            }
+            if constexpr (!is_nothrow_copy_constructible_v<const _Iter&>) {
                 _Mystate = _State::_Nothing;

```

Comment:
This assignment is unneeded for the case that we take the branch above, as we already set `_Mystate`. 

Is it worth dragging that into its own branch `else if (_Right._Mystate == _State::_Holds_iter && _Mystate == _State::_Holds_sentinel)`